applications/mne_sample_set_downloader/downloader.h

```diff
@@ -0,0 +1,141 @@
+* HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+* NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*
+*
+* @brief     Downloader class declaration.
+*
+*/
+
+#ifndef DOWNLOADER_H
+#define DOWNLOADER_H
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// INCLUDES
+//=============================================================================================================
+
+#include "Extract.h"
+
+//*************************************************************************************************************
+//=============================================================================================================
+// QT INCLUDES
+//=============================================================================================================
+
+#include <QObject>
+#include <QMainWindow>
+#include <QNetworkAccessManager>
+#include <QNetworkReply>
+#include <QUrl>
+#include <QString>
+#include <QtCore>
+#include <QFile>
+#include <QSharedPointer>
+#include <QDebug>
+#include <QIODevice>
+#include <QFileDialog>
+
+//*************************************************************************************************************
+//=============================================================================================================
+// DEFINE NAMESPACE DOWNLOADER
+//=============================================================================================================
+
+using namespace std;
+
+namespace Ui {
+class Downloader;
+}
+
+/**
+* This class builds a Downloader Object an GUI. It downloads the sample data set and stores it in a temporary file
+* for the Extract class to extract once it recieves the download_finished signal.
+* The temporary file is deleted on leaving the GUI. The class also handles error and finalisation signals
+* from the extract class and adjusts the GUI accordingly (i.e. displays "Done" on finalization and ask if/where 7zip is
+* installed if the Extract class can't find it).
+*
+* @brief Downloads the sample data set and manages the GUI
+*/
+
+class Downloader : public QMainWindow
+{
+    Q_OBJECT
+
+public:
+    typedef QSharedPointer<Downloader> SPtr;            /**< Shared pointer type for Downloader. */
+    typedef QSharedPointer<const Downloader> ConstSPtr; /**< Const shared pointer type for Downloader. */
+
+    //=========================================================================================================
+    /**
+    * Constructs a Downloader object.
+    */
+    explicit Downloader(QWidget *parent = 0);
+    bool boolDownloadStatus = false;
+    ~Downloader();
+
+public slots:
```

Comment:
If left out the slots for the buttons won't be recognized as such. This causes the buttons on the GUI to lose their function